interfaces/repo.go

```diff
@@ -425,16 +478,15 @@ func (r *Repository) Connected(snapName, plugOrSlotName string) ([]ConnRef, erro
 	r.m.Lock()
 	defer r.m.Unlock()
 
+	return r.connected(snapName, plugOrSlotName)
+}
+
+func (r *Repository) connected(snapName, plugOrSlotName string) ([]ConnRef, error) {
 	if snapName == "" {
-		// Look up the core snap if no snap name was given
-		switch {
-		case r.slots["core"] != nil:
-			snapName = "core"
-		case r.slots["ubuntu-core"] != nil:
-			snapName = "ubuntu-core"
-		default:
-			return nil, fmt.Errorf("snap name is empty")
-		}
+		snapName, _ = r.guessCoreSnapName()
+	}
+	if snapName == "" {
```

Comment:
Sorry, I didn't explain myself properly, I mean that from https://github.com/zyga/snapd/blob/33019a1b6efde8a1a03ee8d514c1c11c18bfe982/interfaces/repo.go#L511, if `snapName` value is not "" when the method is called, it is checked twice against "". IMO the second `if snapName == ""` could be put nested in the first one, after `snapName, _ = r.guessCoreSnapName()`, and maybe check for a not nil error instead of empty snapName, anyway I might be missing something :)